### ISSUE TITLE
KeyError in `PlugManager.provide_plugs`

### ISSUE THREAD

**Issue by kehrazy** [google/openhtf#1096]
```
    def provide_plugs(
        self, plug_name_map: Iterable[Tuple[Text, Type[base_plugs.BasePlug]]]
    ) -> Dict[Text, base_plugs.BasePlug]:
        """Provide the requested plugs [(name, type),] as {name: plug instance}."""
        return {name: self._plugs_by_type[cls] for name, cls in plug_name_map}
```

raises 
> KeyError: <class 'plugs.`plug_name`'>.

Any help?

**Comment by glados-verma** [comment]
What does your phase description looks like? Have you used `with_plugs` with the same keyword as in the phase function?

**Comment by kehrazy** [comment]
Correct me if I'm wrong, by "phase description" you mean something like:

```python
@htf.monitors('measurement_one', monitor_one)
@htf.monitors('measurement_two', monitor_two)
@htf.plugs(some_plug=SomePlug)
def simple_phase(test: htf.TestApi, some_plug: SomePlug):
    ...
```

This raises a KeyError. However, this is not a stable error - I can not seem to get a stable reproducible example.

**Comment by kehrazy** [comment]
If I reverse the `@monitors` order, like this:

```python
@htf.monitors('measurement_two', monitor_two)
@htf.monitors('measurement_one', monitor_one)
@htf.plugs(some_plug=SomePlug)
def simple_phase(test: htf.TestApi, some_plug: SomePlug):
    ...
```

then the second `@monitors` decorator raises the error.

**Comment by kehrazy** [comment]
Pretty sure I've found the issue.

The `@monitors` decorator uses a `@plugs.plug` in the `monitored_phase_func` wrapper.

When running the unmodified code, the behaviour is something like:
```
[ for 'monitor_one': new plug -> result -> "update_kwargs" ] -> False
[ for 'monitor_two': new plug -> result -> "update_kwargs" ] -> False

hello from monitor one!

Exception in thread MonitorTwoThread:
   ...
```

which for some reason makes the `TestExecutor` ignore the required plugs.

If I force the `update_kwargs` to True, the TestExecutor runs just fine, as expected. 
```
[ for 'monitor_one': new plug -> result -> "update_kwargs" ] -> True
[ for 'monitor_two': new plug -> result -> "update_kwargs" ] -> True

hello from monitor one!
hello from monitor two!
```
Is this intended?

**Comment by kehrazy** [comment]
Should be closed with #1099.

**Comment by marcovannoord** [comment]
I can confirm that it works again now, thanks!
